scripts/eurostat/health_determinants/tobacco_consumption/process.py

```diff
@@ -0,0 +1,847 @@
+    return df
+
+
+def daily_smokers_cigarettes_history_income_quintile(
+        df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Cleans the file hlth_ehis_pe1e for concatenation in Final CSV
+    Input Taken: DF
+    Output Provided: DF
+    """
+    cols = [
+        'sex,age,quant_inc,time', 'BE', 'BG', 'CZ', 'DE', 'EE', 'EL', 'ES',
+        'CY', 'LV', 'HU', 'MT', 'AT', 'PL', 'RO', 'SI', 'SK'
+    ]
+    df.columns = cols
+    col1 = "sex,age,quant_inc,time"
+    df = _split_column(df, col1)
+    # Filtering out the wanted rows and columns
+    df = df[(df['age'] == 'TOTAL') & (~(df['quant_inc'] == 'UNK'))]
+    df = _replace_quant_inc(df)
+    df = _replace_sex(df)
+    df.drop(columns=['age'], inplace=True)
+    df['SV'] = 'Percent_Daily_TobaccoSmoking'+\
+        '_Cigarettes'+\
+        '_In_Count_Person_'+df['sex']+'_'+df['quant_inc']
+    df.drop(columns=['quant_inc', 'sex'], inplace=True)
+    df = df.melt(id_vars=['SV','time'], var_name='geo'\
+        ,value_name='observation')
+    return df
+
+
+def dsmokers_number_of_cigarettes_history_education_attainment_level(
+        df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Cleans the file hlth_ehis_pe1e for concatenation in Final CSV
+    Input Taken: DF
+    Output Provided: DF
+    """
+    cols = [
+        'sex,age,smoking,isced11,time', 'BE', 'BG', 'CZ', 'DE', 'EE', 'EL',
+        'ES', 'CY', 'LV', 'HU', 'MT', 'PL', 'RO', 'SI', 'SK'
+    ]
+    df.columns = cols
+    col1 = "sex,age,smoking,isced11,time"
+    df = _split_column(df, col1)
+    # Filtering out the wanted rows and columns
+    df = df[df['age'] == 'TOTAL']
+    df = _replace_isced11(df)
+    df = _replace_smoking(df)
+    df = _replace_sex(df)
+    df.drop(columns=['age'], inplace=True)
+    df['SV'] = 'Percent_Daily'+'_'+df['smoking']+'_TobaccoSmoking'+\
+        '_Cigarettes'+\
+        '_In_Count_Person_'+df['isced11']+'_'+df['sex']
+    df.drop(columns=['smoking', 'isced11', 'sex'], inplace=True)
+    df = df.melt(id_vars=['SV','time'], var_name='geo'\
+        ,value_name='observation')
+    return df
+
+
+class EuroStatTobaccoConsumption:
+    """
+    This Class has requried methods to generate Cleaned CSV,
+    MCF and TMCF Files.
+    """
+
+    def __init__(self, input_files: list, csv_file_path: str,\
+        mcf_file_path: str, tmcf_file_path: str) -> None:
+        self.input_files = input_files
+        self.cleaned_csv_file_path = csv_file_path
+        self.mcf_file_path = mcf_file_path
+        self.tmcf_file_path = tmcf_file_path
+
+    def _generate_tmcf(self) -> None:
+        """
+        This method generates TMCF file w.r.t
+        dataframe headers and defined TMCF template.
+        Arguments:
+            None
+        Returns:
+            None
+        """
+        # Writing Genereated TMCF to local path.
+        with open(self.tmcf_file_path, 'w+', encoding="UTF-8") as f_out:
+            f_out.write(_TMCF_TEMPLATE.rstrip('\n'))
+
+    def _generate_mcf(self, sv_list) -> None:
```

Comment:
Can some of the logic in this function go in the common folder where you have a similar function to generate the mcf for the denominator? I think it might be good to have these two mcf generators and formatters in the same file and call the necessary ones from here.